src/com/rcv/Tabulator.java

```diff
@@ -207,6 +206,60 @@ void tabulate() {
           candidateToRoundEliminated.put(loser, currentRound);
         }
       }
+
+      // TODO: avoid recomputing these every round (since they should be static for a given
+      // candidate once that candidate has won)?
+      updateWinnerTallies();
+    }
+  }
+
+  // function: updateWinnerTallies
+  // purpose: update the tally for the just-completed round to reflect the tallies for candidates
+  // who won in a past round
```

Comment:
Looks good.  Could you please add a bit more comments on why we need to update past winners?  I know the context so I get it, but to anyone else trying to navigate this code it would be helpful.   Also (annoyingly) the pastWinners local variable below needs an inline comment too.  Thanks.

Reply:
Will do.